datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala

```diff
@@ -127,6 +136,45 @@ object DatasetComparisonJob {
     }
   }
 
+  private def handleDuplicates(outPath: String,
+                               dfsWithKey: GenericPair[DataFrame]): (GenericPair[Long], GenericPair[DataFrame]) = {
+    def writeDown(df: DataFrame, duplicates: DataFrame, path: String): Unit = {
+      df.alias("original")
+        .join(duplicates, Seq(comparisonUniqueId), "left_outer")
```

Comment:
You're not removing the `comparisonUniqueId` from the output.

Reply:
I think you need an `inner join` here since the `left outer join` here will give you the original dataset.